aiida/orm/nodes/data/code/legacy.py

```diff
@@ -48,6 +41,16 @@ class Code(AbstractCode):
     # pylint: disable=too-many-public-methods
 
     def __init__(self, remote_computer_exec=None, local_executable=None, input_plugin_name=None, files=None, **kwargs):
+
+        if self.__class__ == Code:
```

Comment:
or actually, since there is deprectation warnings below, for non-null `remote_computer_exec`/`local_executable`, maybe this deprecation warning does not need to be here at all?

Reply:
There is a problem with this solution, because the entire class is deprecated, not just instances of it. There is quite a bit of code out there that never explicitly constructs new instances but they just use the class, for example code like this:
```python
from aiida.orm import Code
code = load_code('some-name')
assert isinstance(code, Code)
```
or when they declare ports in processes and set `valid_type=Code`.

We should find a way to have the deprecation warning also be emitted in these cases.

I thought of applying the usual trick of having the actual deprecated code to `_legacy` and internally use that (we still need to use it because the `InstalledCode` need subclass `Code` as to not break backwards-compatibility) so that we don't hit the deprecation warning unnecessarily. But the problem is that the `Code` class was importable directly from `aiida.orm`, so we should still import it there, which will therefore always trigger the deprecation warning on import. Even if you import some other code class. I wasn't able to find a better solution so just left this as is.

Reply:
maybe you can catch+ignore the warning, when you import internally